src/tools/MonarchPeasantSample/main.cpp

```diff
@@ -0,0 +1,175 @@
+﻿#include "pch.h"
+#include <conio.h>
+#include "Monarch.h"
+#include "Peasant.h"
+#include "AppState.h"
+#include "../../types/inc/utils.hpp"
+
+using namespace winrt;
+using namespace winrt::Windows::Foundation;
+using namespace ::Microsoft::Console;
+
+////////////////////////////////////////////////////////////////////////////////
+// This seems like a hack, but it works.
+//
+// This class factory works so that there's only ever one instance of a Monarch
+// per-process. Once the first monarch is created, we'll stash it in g_weak.
+// Future callers who try to instantiate a Monarch will get the one that's
+// already been made.
+//
+// I'm sure there's a better way to do this with WRL, but I'm not familiar
```

Comment:
With WRL it's 
`Microsoft::WRL::CoCreateableClass(yourclassname)` in the header so it will set up the goo in the library.

And then you have to call `Module<T>::Create().RegisterObjects()` to make it walk through that goo and make simple factories and register them for you.

However, I don't really recommend that as it has a lot of magic that can go wrong that is hard to debug specifically when you start mixing COM and WinRT, InProc and OutOfProc, and STA and MTA.

However, you **could** potentially use `Microsoft::WRL::SimpleClassFactory<T>()` here instead of implementing it  yourself as `MonarchFactory` by doing
`make<Microsoft::WRL::SimpleClassFactory<MonarchPeasantSample::implementation::Monarch>>()` down in the `CoRegisterClassObject`
as far as I'm aware.

Reply:
So, that's what I wanted to do, yes. 

There's a catch though, I want the `Monarch` to be a singleton object per-process. So there's only just one instance of it in a process. I'm under the impression that the WRL goo is gonna wire it up so every `CoCreateInstance` gets a new one.

Reply:
That's probably right, yes. OK.